src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -33,6 +33,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       could be removed by the ILLink, causing a trimmed app to crash. -->
   <PropertyGroup Condition="'$(PublishTrimmed)' == 'true'">
     <StartupHookSupport Condition="'$(StartupHookSupport)' == ''">false</StartupHookSupport>
+    <MetadataUpdaterSupport Condition="'$(MetadataUpdaterSupport)' == '' and ('$(DebuggerSupport)' != 'true' or '$(Configuration)' != 'Debug')">false</MetadataUpdaterSupport>
```

Comment:
This looks partially redundant with https://github.com/dotnet/sdk/blob/535f8402195ba520ffe4671bd30b29b403b51c7d/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L153

But I don't understand why we condition this at all. Chances of being able to do metadata updates in trimmed apps are slim and I'm not aware of any tooling that would even try. I think this should be just unconditionally set to `false`.

There was a discussion of doing that in https://github.com/dotnet/sdk/pull/18810#discussion_r666525265 but it wasn't done because there was a fear it would affect F5 time as well. I can't find any F5-time place that would be affected by this. Am I missing something? Cc @pranavkm @eerhardt @sbomer

Reply:
> `dotnet watch` doesn't run ILLink, right ?

It doesn't. The problem with metadata updater and trimming is that trimming performs whole program analysis and so a small change somewhere can have butterfly effects elsewhere in the program. Applying such updates doesn't seem like a trivial task (ILLink might need to be able to generate additional updates?) and I don't think any customer has ever asked for that. We don't use trimming in inner loop.

Reply:
Actually we do unit test trimming+hot-reload+Release.
[Log](https://helixr1107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-111215-merge-fdd98f541cbc4aed95/WasmTestOnChrome-ST-System.Runtime.Loader.Tests/1/console.c6836af1.log?helixlogtype=result)

I think that should be fixed by applying `<MetadataUpdaterSupport>false</MetadataUpdaterSupport>` to the unit test.

Reply:
I moved these changes to separate PR https://github.com/dotnet/runtime/pull/111247